src/main/scala/com/github/saurfang/sas/spark/SasRelation.scala

```diff
@@ -16,119 +16,251 @@
- */
+  * Defines a RDD that is backed by [[SasInputFormat]]. Data are coerced into appropriate types according to
+  * meta information embedded in .sas7bdat file.
+  */
 case class SasRelation protected[spark](
                                          location: String,
                                          userSchema: StructType = null,
-                                         @transient conf: JobConf = new JobConf(),
-                                         minPartitions: Int = 0
-                                         )(@transient val sqlContext: SQLContext)
-  extends BaseRelation with TableScan {
+                                         extractLabel: Boolean = false,
+                                         forceLowercaseNames: Boolean = false,
+                                         inferDecimal: Boolean = false,
+                                         inferDecimalScale: Option[Int] = None,
+                                         inferFloat: Boolean = false,
+                                         inferInt: Boolean = false,
+                                         inferLong: Boolean = false,
+                                         inferShort: Boolean = false,
+                                         metadataTimeout: Int = 60,
+                                         minSplitSize: Option[Long] = None,
+                                         maxSplitSize: Option[Long] = None
+                                       )(@transient val sqlContext: SQLContext) extends BaseRelation with TableScan {
+
   @transient lazy val log = LogManager.getLogger(this.getClass.getName)
-  val schema = inferSchema()
-
-  // By making this a lazy val we keep the RDD around, amortizing the cost of locating splits.
-  def buildScan(): RDD[Row] = {
-    FileInputFormat.setInputPaths(conf, new Path(location))
-    val baseRDD = new HadoopRDD[NullWritable, Array[Object]](
-      sqlContext.sparkContext,
-      conf,
-      classOf[SasInputFormat],
-      classOf[NullWritable],
-      classOf[Array[Object]],
-      minPartitions
-    ).map(_._2)
-
-    baseRDD.mapPartitions { iter => parseSAS(iter, schema.fields) }
+
+  // Automatically extract schema from file.
+  val schema = inferSchema(
+    extractLabel = extractLabel,
+    forceLowercaseNames = forceLowercaseNames,
+    inferDecimal = inferDecimal,
+    inferDecimalScale = inferDecimalScale,
+    inferFloat = inferFloat,
+    inferInt = inferInt,
+    inferLong = inferLong,
+    inferShort = inferShort,
+    metadataTimeout = metadataTimeout
+  )
+
+  override def buildScan: RDD[Row] = {
+
+    // Start from the hadoopConfiguration in sparkContext.
+    val conf: Configuration = new Configuration(sqlContext.sparkContext.hadoopConfiguration)
+
+    // Set min/max split sizes if provided.
+    if (!minSplitSize.isEmpty) {conf.setLong("mapred.min.split.size", minSplitSize.get)}
+    if (!maxSplitSize.isEmpty) {conf.setLong("mapred.max.split.size", maxSplitSize.get)}
```

Comment:
`mapred.min.split.size` is deprecated in favor of `mapreduce.input.fileinputformat.split.minsize`.

https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-common/DeprecatedProperties.html

https://hadoop.apache.org/docs/r2.6.2/hadoop-project-dist/hadoop-common/DeprecatedProperties.html

Same for `mapred.max.split.size`.

Reply:
I don't think we should expose these two as parameters if all we do is to pass them down as Spark configs. It is fine to educate users about these configurations in the readme. AFAIK no other data source in Spark have these kind of proxy configs.

Reply:
my 1.5 cents - Good point. On the other hand, `spark-sas7bdat` is tightly coupled with hadoop already .. I agree, that moving this just to docs makes sense, as there is a way to set same configs through spark config. 

ps. Not every Spark user runs Spark on Hadoop.. might be a good idea to refactor it in the future to make `spark-sas7bdat` more abstract so somebody could use it for Spark-on-Kubernetes, for example. I think the splitting logic would be the same for `S3` storage for example?

Reply:
@Tagar I am not 100% sure, but the current code should support S3 URI's already? As spark uses hadoop code for reading anyway, even if running on K8S.

@saurfang, it's up to you, but the reason I include them here, is because people might want to specify max/min size at the file level, rather than globally for their spark context. (Notice that I make a copy of the `sparkContext.hadoopConfiguration` before adding these configs, rather than changing the users global conf)